### PR TITLE
Remove .gitignore file from dist package

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Not a big deal, but `.gitignore` file is present in dist package fetched from packagist.

### CODE REVIEW

**Comment by cedric-anne** [comment]
Sorry for the multiple commits, I have made it one by one from GitHub UI. I can squash them if it is a problem.

**Review by GromNaN — APPROVED** [review]
> Sorry for the multiple commits, I have made it one by one from GitHub UI. I can squash them if it is a problem.

Don't worry, we have a tool to squash before merging.

**Comment by nicolas-grekas** [comment]
There are more of them:
```
src/Symfony/Contracts/Cache/.gitignore
src/Symfony/Contracts/.gitignore
src/Symfony/Contracts/EventDispatcher/.gitignore
src/Symfony/Contracts/Service/.gitignore
src/Symfony/Contracts/Translation/.gitignore
src/Symfony/Contracts/Deprecation/.gitignore
src/Symfony/Contracts/HttpClient/.gitignore
```

**Comment by cedric-anne** [comment]
I added `.gitattributes` files where they were missing, and also added `export-ignore` on test files.
By the way, I noticed that some the namespace of test files is sometimes `Test` on `Symfony\Contracts`, and that is not consistent with namespace `Tests` used everywhere else in Symfony codebase.

**Comment by cedric-anne** [comment]
> By the way, I noticed that some the namespace of test files is sometimes `Test` on `Symfony\Contracts`, and that is not consistent with namespace `Tests` used everywhere else in Symfony codebase.

My bad, these should not be excluded.

File: src/Symfony/Contracts/Translation/.gitattributes [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/Test export-ignore
```
Comment (fabpot) [inline src/Symfony/Contracts/Translation/.gitattributes]: Should be reverted. The `Test` namespace contains files that people can reuse in their own tests. The `Tests` namespace contains tests that we can indeed ignore.

**Review by fabpot — CHANGES_REQUESTED** [review]

**Review by GromNaN — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @cedric-anne.

**Pull request merged**